scripts/check-file-changes.sh

```diff
@@ -2,6 +2,6 @@
 
 BRANCH="$(git rev-parse --abbrev-ref HEAD)"
 echo $BRANCH
-if [[ "$BRANCH" != "master" ]]; then
-    git diff master --name-only --no-color | egrep -e "^(\.circleci\/.*)|(.*\.(go|mod|sum))|Makefile$" || circleci step halt;
+if [[ "$BRANCH" != "master" ]] && [[ "$BRANCH" != release* ]]; then
+    git diff master --name-only --no-color | egrep -e "^(\.circleci\/.*)|(.*\.(go|mod|sum|sh))|Makefile$" || circleci step halt;
```

Comment:
Thanks for adding the check for release branches!

We ask git for all the differences between this branch and master, then use grep to look for changes in the .circleci/ directory, anything named `*.go*` or `*.mod*` or `*.sum*` or `*.sh*` or `*Makefile`. I think there's an operator precedence error here with how ^$ and | work. You probably need ^$ in each alternate section. The first ^ only applies to .cirleci/.* and the last $ only applies to Makefile.

Why do we filter the differences with grep? Is it to avoid builds for doc-only changes? Could you add a comment so it will be more obvious?